rmsutil/system.py

```diff
@@ -0,0 +1,31 @@
+import contextlib
+import copy
+import os
+import sys
+
+
+@contextlib.contextmanager
+def push_env(**kwargs):
+    try:
+        old_env = os.environ.copy()
```

Comment:
This probably doesn't do what you think it does, because `os.environ` is an instance of a subclass of dict, and subclassing is of course anathema, no good, super bad, etc.

Hint: `os.environ` is an `instance` (yep, really), `os.environ.copy()` is a `dict`. If you want to build your own environ, you might want to consider `os._Environ` because that's what `os.environ` is actually an instance of.